Signal/src/Jobs/MarkIdentityAsSeenJob.swift

```diff
@@ -8,18 +8,24 @@ import Foundation
 class MarkIdentityAsSeenJob: NSObject {
     let TAG = "[MarkIdentityAsSeenJob]"
 
-    private let thread: TSThread
+    private let recipientIds: [String]
 
     public class func run(thread: TSThread) {
-        MarkIdentityAsSeenJob(thread: thread).run()
+        let recipientIds = thread.recipientIdentifiers
+
+        MarkIdentityAsSeenJob(recipientIds: recipientIds).run()
+    }
+
+    public class func run(recipientId: String) {
+        MarkIdentityAsSeenJob(recipientIds: [recipientId]).run()
     }
 
-    init(thread: TSThread) {
-        self.thread = thread
+    init(recipientIds: [String]) {
+        self.recipientIds = recipientIds
     }
 
     public func run() {
-        for recipientId in self.thread.recipientIdentifiers {
+        for recipientId in self.recipientIds {
```

Comment:
I wonder if we should modify all of the "jobs" to dispatch async to the global worker thread to ensure this kind of thing is never done on the main thread.